Memory leaks in all test running "make check"

I compiled i3status, without any apparent issues, when I run "make check", i get the following output. I am using the last commit 5aec4a5da32e9a1fad1a89f17b10d676d4312895 from git repository.

Can someone try to reproduce it to see if it's something about my build, or it is a real issue ?
Thanks

 ```
=====================================
   i3status 2.13: ./test-suite.log
=====================================

# TOTAL: 1
# PASS:  0
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0

.. contents:: :depth: 2

FAIL: travis/run-tests.pl
=========================


=================================================================
==7423==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7f4fdfff3330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7f4fdfef388e in yajl_gen_alloc (/lib/x86_64-linux-gnu/libyajl.so.2+0x588e)
    #2 0x7f4fdfef522f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7f4fdfff3330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7f4fdfef2d5e  (/lib/x86_64-linux-gnu/libyajl.so.2+0x4d5e)
    #2 0x7f4fdfef522f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '001-battery'… Crash!

=================================================================
==7430==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7f4477567330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7f447746788e in yajl_gen_alloc (/lib/x86_64-linux-gnu/libyajl.so.2+0x588e)
    #2 0x7f447746922f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7f4477567330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7f4477466d5e  (/lib/x86_64-linux-gnu/libyajl.so.2+0x4d5e)
    #2 0x7f447746922f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '007-cpu-usage-degraded-threshold-format'… Crash!

=================================================================
==7439==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7fbb80db7330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7fbb80cb788e in yajl_gen_alloc (/lib/x86_64-linux-gnu/libyajl.so.2+0x588e)
    #2 0x7fbb80cb922f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7fbb80db7330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7fbb80cb6d5e  (/lib/x86_64-linux-gnu/libyajl.so.2+0x4d5e)
    #2 0x7fbb80cb922f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '003-path-exists'… Crash!

=================================================================
==7446==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7f23ff58a330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7f23ff48a88e in yajl_gen_alloc (/lib/x86_64-linux-gnu/libyajl.so.2+0x588e)
    #2 0x7f23ff48c22f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7f23ff58a330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7f23ff489d5e  (/lib/x86_64-linux-gnu/libyajl.so.2+0x4d5e)
    #2 0x7f23ff48c22f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '019-battery-capacity'… Crash!

=================================================================
==7456==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7f910be62330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7f910bd6288e in yajl_gen_alloc (/lib/x86_64-linux-gnu/libyajl.so.2+0x588e)
    #2 0x7f910bd6422f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7f910be62330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7f910bd61d5e  (/lib/x86_64-linux-gnu/libyajl.so.2+0x4d5e)
    #2 0x7f910bd6422f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '015-battery-capacity'… Crash!

=================================================================
==7461==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7ff669cd8330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7ff669bd888e in yajl_gen_alloc (/lib/x86_64-linux-gnu/libyajl.so.2+0x588e)
    #2 0x7ff669bda22f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7ff669cd8330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7ff669bd7d5e  (/lib/x86_64-linux-gnu/libyajl.so.2+0x4d5e)
    #2 0x7ff669bda22f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '018-battery-capacity'… Crash!

=================================================================
==7466==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7f07a4daa330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7f07a4caa88e in yajl_gen_alloc (/lib/x86_64-linux-gnu/libyajl.so.2+0x588e)
    #2 0x7f07a4cac22f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7f07a4daa330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7f07a4ca9d5e  (/lib/x86_64-linux-gnu/libyajl.so.2+0x4d5e)
    #2 0x7f07a4cac22f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '009-disk-low-threshold-format'… Crash!
i3status: provided CPU number '10' above detected number of CPU 4

=================================================================
==7474==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7f1fcd40b330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7f1fcd30b88e in yajl_gen_alloc (/lib/x86_64-linux-gnu/libyajl.so.2+0x588e)
    #2 0x7f1fcd30d22f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7f1fcd40b330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7f1fcd30ad5e  (/lib/x86_64-linux-gnu/libyajl.so.2+0x4d5e)
    #2 0x7f1fcd30d22f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '022-cpu-usage-tenth-cpu'… Crash!

=================================================================
==7484==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7fe2aa337330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7fe2aa23788e in yajl_gen_alloc (/lib/x86_64-linux-gnu/libyajl.so.2+0x588e)
    #2 0x7fe2aa23922f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7fe2aa337330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7fe2aa236d5e  (/lib/x86_64-linux-gnu/libyajl.so.2+0x4d5e)
    #2 0x7fe2aa23922f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '011-cpu-usage'… Crash!

=================================================================
==7491==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7f4350e0e330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7f4350d0e88e in yajl_gen_alloc (/lib/x86_64-linux-gnu/libyajl.so.2+0x588e)
    #2 0x7f4350d1022f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7f4350e0e330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7f4350d0dd5e  (/lib/x86_64-linux-gnu/libyajl.so.2+0x4d5e)
    #2 0x7f4350d1022f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '002-cpu-temp'… Crash!

=================================================================
==7496==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7f646956c330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7f646946c88e in yajl_gen_alloc (/lib/x86_64-linux-gnu/libyajl.so.2+0x588e)
    #2 0x7f646946e22f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7f646956c330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7f646946bd5e  (/lib/x86_64-linux-gnu/libyajl.so.2+0x4d5e)
    #2 0x7f646946e22f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '021-battery-second'… Crash!
i3status: Cannot read CPU usage

=================================================================
==7501==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7f0ec669a330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7f0ec659a88e in yajl_gen_alloc (/lib/x86_64-linux-gnu/libyajl.so.2+0x588e)
    #2 0x7f0ec659c22f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7f0ec669a330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7f0ec6599d5e  (/lib/x86_64-linux-gnu/libyajl.so.2+0x4d5e)
    #2 0x7f0ec659c22f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '012-cpu-usage-error'… Crash!

=================================================================
==7506==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7f8f6a016330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7f8f69f1688e in yajl_gen_alloc (/lib/x86_64-linux-gnu/libyajl.so.2+0x588e)
    #2 0x7f8f69f1822f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7f8f6a016330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7f8f69f15d5e  (/lib/x86_64-linux-gnu/libyajl.so.2+0x4d5e)
    #2 0x7f8f69f1822f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '008-load-max-threshold-format'… Crash!

=================================================================
==7513==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7fc343d83330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7fc343c8388e in yajl_gen_alloc (/lib/x86_64-linux-gnu/libyajl.so.2+0x588e)
    #2 0x7fc343c8522f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7fc343d83330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7fc343c82d5e  (/lib/x86_64-linux-gnu/libyajl.so.2+0x4d5e)
    #2 0x7fc343c8522f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '016-ethernet-down'… Crash!

=================================================================
==7523==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7f6f211be330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7f6f210be88e in yajl_gen_alloc (/lib/x86_64-linux-gnu/libyajl.so.2+0x588e)
    #2 0x7f6f210c022f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7f6f211be330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7f6f210bdd5e  (/lib/x86_64-linux-gnu/libyajl.so.2+0x4d5e)
    #2 0x7f6f210c022f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '025-file-contents'… Crash!
i3status: provided CPU number cannot be parsed
=================================================================
==7528==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x60200000319a at pc 0x561cbbf17dbd bp 0x7ffcf0b2d050 sp 0x7ffcf0b2d048
READ of size 1 at 0x60200000319a thread T0
    #0 0x561cbbf17dbc in print_cpu_usage (/home/guiu/source/i3status/build/i3status+0x28dbc)
    #1 0x561cbbf0ee3c in main (/home/guiu/source/i3status/build/i3status+0x1fe3c)
    #2 0x7f4941eb209a in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x2409a)
    #3 0x561cbbeffdf9 in _start (/home/guiu/source/i3status/build/i3status+0x10df9)

0x60200000319a is located 0 bytes to the right of 10-byte region [0x602000003190,0x60200000319a)
allocated by thread T0 here:
    #0 0x7f49427d1810 in strdup (/lib/x86_64-linux-gnu/libasan.so.5+0x3a810)
    #1 0x7f494278d4ad in cfg_setopt (/lib/x86_64-linux-gnu/libconfuse.so.2+0x54ad)

SUMMARY: AddressSanitizer: heap-buffer-overflow (/home/guiu/source/i3status/build/i3status+0x28dbc) in print_cpu_usage
Shadow bytes around the buggy address:
  0x0c047fff85e0: fa fa 00 fa fa fa 00 fa fa fa 00 fa fa fa 00 fa
  0x0c047fff85f0: fa fa 00 fa fa fa 00 fa fa fa 00 fa fa fa 00 fa
  0x0c047fff8600: fa fa 00 fa fa fa 00 fa fa fa 00 fa fa fa 00 fa
  0x0c047fff8610: fa fa 00 fa fa fa 00 fa fa fa 05 fa fa fa 00 fa
  0x0c047fff8620: fa fa 00 fa fa fa 00 02 fa fa 00 fa fa fa 00 fa
=>0x0c047fff8630: fa fa 00[02]fa fa 00 fa fa fa fa fa fa fa fa fa
  0x0c047fff8640: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c047fff8650: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c047fff8660: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c047fff8670: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c047fff8680: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07 
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
==7528==ABORTING
Testing test case '024-cpu-usage-invalid-cpu'… Crash!

=================================================================
==7531==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7f1ed2a50330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7f1ed295088e in yajl_gen_alloc (/lib/x86_64-linux-gnu/libyajl.so.2+0x588e)
    #2 0x7f1ed295222f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7f1ed2a50330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7f1ed294fd5e  (/lib/x86_64-linux-gnu/libyajl.so.2+0x4d5e)
    #2 0x7f1ed295222f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '017-wireless-down'… Crash!

=================================================================
==7536==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7ff23a82e330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7ff23a72e88e in yajl_gen_alloc (/lib/x86_64-linux-gnu/libyajl.so.2+0x588e)
    #2 0x7ff23a73022f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7ff23a82e330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7ff23a72dd5e  (/lib/x86_64-linux-gnu/libyajl.so.2+0x4d5e)
    #2 0x7ff23a73022f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '005-cpu-temp-max-threshold-format'… Crash!

=================================================================
==7541==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7fa2e0eff330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7fa2e0dff88e in yajl_gen_alloc (/lib/x86_64-linux-gnu/libyajl.so.2+0x588e)
    #2 0x7fa2e0e0122f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7fa2e0eff330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7fa2e0dfed5e  (/lib/x86_64-linux-gnu/libyajl.so.2+0x4d5e)
    #2 0x7fa2e0e0122f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '006-cpu-usage-max-threshold-format'… Crash!

=================================================================
==7546==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7fc6ff0fd330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7fc6feffd88e in yajl_gen_alloc (/lib/x86_64-linux-gnu/libyajl.so.2+0x588e)
    #2 0x7fc6fefff22f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7fc6ff0fd330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7fc6feffcd5e  (/lib/x86_64-linux-gnu/libyajl.so.2+0x4d5e)
    #2 0x7fc6fefff22f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '004-run-watch'… Crash!
i3status: Cannot read CPU usage

=================================================================
==7551==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7feaa5665330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7feaa556588e in yajl_gen_alloc (/lib/x86_64-linux-gnu/libyajl.so.2+0x588e)
    #2 0x7feaa556722f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7feaa5665330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7feaa5564d5e  (/lib/x86_64-linux-gnu/libyajl.so.2+0x4d5e)
    #2 0x7feaa556722f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '013-cpu-usage-error'… Crash!

=================================================================
==7562==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7fd5956aa330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7fd5955aa88e in yajl_gen_alloc (/lib/x86_64-linux-gnu/libyajl.so.2+0x588e)
    #2 0x7fd5955ac22f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7fd5956aa330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7fd5955a9d5e  (/lib/x86_64-linux-gnu/libyajl.so.2+0x4d5e)
    #2 0x7fd5955ac22f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '023-battery-not-charging'… Crash!

=================================================================
==7573==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7f029b17a330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7f029b07a88e in yajl_gen_alloc (/lib/x86_64-linux-gnu/libyajl.so.2+0x588e)
    #2 0x7f029b07c22f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7f029b17a330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7f029b079d5e  (/lib/x86_64-linux-gnu/libyajl.so.2+0x4d5e)
    #2 0x7f029b07c22f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '014-battery-capacity'… Crash!

=================================================================
==7581==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7fc3b6a9b330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7fc3b699b88e in yajl_gen_alloc (/lib/x86_64-linux-gnu/libyajl.so.2+0x588e)
    #2 0x7fc3b699d22f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7fc3b6a9b330 in __interceptor_malloc (/lib/x86_64-linux-gnu/libasan.so.5+0xe9330)
    #1 0x7fc3b699ad5e  (/lib/x86_64-linux-gnu/libyajl.so.2+0x4d5e)
    #2 0x7fc3b699d22f  (/lib/x86_64-linux-gnu/libyajl.so.2+0x722f)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '010-cpu-usage'… Crash!
FAIL travis/run-tests.pl (exit status: 1)
```

Making i3status-2.13 on fresh ubuntu-18.04LTS image results in the same problem as above.

```
ubuntu@legible-chinook:~/i3status-2.13/build$ make check
make  travis/run-tests.pl
make[1]: Entering directory '/home/ubuntu/i3status-2.13/build'
make[1]: 'travis/run-tests.pl' is up to date.
make[1]: Leaving directory '/home/ubuntu/i3status-2.13/build'
make  check-TESTS
make[1]: Entering directory '/home/ubuntu/i3status-2.13/build'
make[2]: Entering directory '/home/ubuntu/i3status-2.13/build'
FAIL: travis/run-tests.pl
============================================================================
Testsuite summary for i3status 2.13
============================================================================
# TOTAL: 1
# PASS:  0
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0
============================================================================
See ./test-suite.log
Please report to https://github.com/i3/i3status/issues
============================================================================
Makefile:1325: recipe for target 'test-suite.log' failed
make[2]: *** [test-suite.log] Error 1
make[2]: Leaving directory '/home/ubuntu/i3status-2.13/build'
Makefile:1431: recipe for target 'check-TESTS' failed
make[1]: *** [check-TESTS] Error 2
make[1]: Leaving directory '/home/ubuntu/i3status-2.13/build'
Makefile:1636: recipe for target 'check-am' failed
make: *** [check-am] Error 2
```

<details>
<summary>test-suite.log</summary>

```
=====================================
   i3status 2.13: ./test-suite.log
=====================================

# TOTAL: 1
# PASS:  0
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0

.. contents:: :depth: 2

FAIL: travis/run-tests.pl
=========================


=================================================================
==16744==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7f39fc31bb40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7f39fbe2a77e in yajl_gen_alloc (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x477e)
    #2 0x7f39fbe2c0df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7f39fc31bb40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7f39fbe29c3e  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x3c3e)
    #2 0x7f39fbe2c0df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '001-battery'… [1m[31mCrash![0m

=================================================================
==16749==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7f6a34fd3b40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7f6a34ae277e in yajl_gen_alloc (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x477e)
    #2 0x7f6a34ae40df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7f6a34fd3b40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7f6a34ae1c3e  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x3c3e)
    #2 0x7f6a34ae40df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '015-battery-capacity'… [1m[31mCrash![0m
i3status: Cannot read CPU usage

=================================================================
==16754==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7fc254527b40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7fc25403677e in yajl_gen_alloc (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x477e)
    #2 0x7fc2540380df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7fc254527b40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7fc254035c3e  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x3c3e)
    #2 0x7fc2540380df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '013-cpu-usage-error'… [1m[31mCrash![0m
i3status: Cannot read CPU usage

=================================================================
==16759==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7fc481aabb40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7fc4815ba77e in yajl_gen_alloc (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x477e)
    #2 0x7fc4815bc0df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7fc481aabb40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7fc4815b9c3e  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x3c3e)
    #2 0x7fc4815bc0df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '012-cpu-usage-error'… [1m[31mCrash![0m
i3status: provided CPU number cannot be parsed
=================================================================
==16764==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x60200000319a at pc 0x5597f5049f5b bp 0x7ffd7be3a850 sp 0x7ffd7be3a840
READ of size 1 at 0x60200000319a thread T0
    #0 0x5597f5049f5a in print_cpu_usage ../../i3status-2.13/src/print_cpu_usage.c:159
    #1 0x5597f504101e in main ../../i3status-2.13/i3status.c:800
    #2 0x7f341ed69b96 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x21b96)
    #3 0x5597f5031fb9 in _start (/home/ubuntu/i3status-2.13/build/i3status+0xffb9)

0x60200000319a is located 0 bytes to the right of 10-byte region [0x602000003190,0x60200000319a)
allocated by thread T0 here:
    #0 0x7f3420501538 in strdup (/usr/lib/x86_64-linux-gnu/libasan.so.4+0x77538)
    #1 0x7f342028199a in cfg_setopt (/usr/lib/x86_64-linux-gnu/libconfuse.so.2+0x499a)

SUMMARY: AddressSanitizer: heap-buffer-overflow ../../i3status-2.13/src/print_cpu_usage.c:159 in print_cpu_usage
Shadow bytes around the buggy address:
  0x0c047fff85e0: fa fa 00 fa fa fa 00 fa fa fa 00 fa fa fa 00 fa
  0x0c047fff85f0: fa fa 00 fa fa fa 00 fa fa fa 00 fa fa fa 00 fa
  0x0c047fff8600: fa fa 00 fa fa fa 00 fa fa fa 00 fa fa fa 00 fa
  0x0c047fff8610: fa fa 00 fa fa fa 00 fa fa fa 05 fa fa fa 00 fa
  0x0c047fff8620: fa fa 00 fa fa fa 00 02 fa fa 00 fa fa fa 00 fa
=>0x0c047fff8630: fa fa 00[02]fa fa 00 fa fa fa fa fa fa fa fa fa
  0x0c047fff8640: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c047fff8650: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c047fff8660: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c047fff8670: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c047fff8680: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07 
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
==16764==ABORTING
Testing test case '024-cpu-usage-invalid-cpu'… [1m[31mCrash![0m

=================================================================
==16767==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7f9981707b40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7f998121677e in yajl_gen_alloc (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x477e)
    #2 0x7f99812180df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7f9981707b40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7f9981215c3e  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x3c3e)
    #2 0x7f99812180df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '019-battery-capacity'… [1m[31mCrash![0m

=================================================================
==16772==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7f691d284b40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7f691cd9377e in yajl_gen_alloc (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x477e)
    #2 0x7f691cd950df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7f691d284b40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7f691cd92c3e  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x3c3e)
    #2 0x7f691cd950df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '014-battery-capacity'… [1m[31mCrash![0m

=================================================================
==16777==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7fcc7c787b40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7fcc7c29677e in yajl_gen_alloc (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x477e)
    #2 0x7fcc7c2980df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7fcc7c787b40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7fcc7c295c3e  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x3c3e)
    #2 0x7fcc7c2980df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '006-cpu-usage-max-threshold-format'… [1m[31mCrash![0m

=================================================================
==16782==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7f401236ab40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7f4011e7977e in yajl_gen_alloc (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x477e)
    #2 0x7f4011e7b0df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7f401236ab40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7f4011e78c3e  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x3c3e)
    #2 0x7f4011e7b0df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '003-path-exists'… [1m[31mCrash![0m

=================================================================
==16787==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7f2dcd860b40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7f2dcd36f77e in yajl_gen_alloc (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x477e)
    #2 0x7f2dcd3710df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7f2dcd860b40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7f2dcd36ec3e  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x3c3e)
    #2 0x7f2dcd3710df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '025-file-contents'… [1m[31mCrash![0m

=================================================================
==16792==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7f2f4242db40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7f2f41f3c77e in yajl_gen_alloc (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x477e)
    #2 0x7f2f41f3e0df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7f2f4242db40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7f2f41f3bc3e  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x3c3e)
    #2 0x7f2f41f3e0df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '018-battery-capacity'… [1m[31mCrash![0m

=================================================================
==16797==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7fbcbeed7b40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7fbcbe9e677e in yajl_gen_alloc (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x477e)
    #2 0x7fbcbe9e80df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7fbcbeed7b40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7fbcbe9e5c3e  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x3c3e)
    #2 0x7fbcbe9e80df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '016-ethernet-down'… [1m[31mCrash![0m
i3status: provided CPU number '10' above detected number of CPU 1

=================================================================
==16805==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7f8504322b40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7f8503e3177e in yajl_gen_alloc (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x477e)
    #2 0x7f8503e330df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7f8504322b40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7f8503e30c3e  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x3c3e)
    #2 0x7f8503e330df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '022-cpu-usage-tenth-cpu'… [1m[31mCrash![0m
i3status: provided CPU number '1' above detected number of CPU 1

=================================================================
==16815==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7ff162891b40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7ff1623a077e in yajl_gen_alloc (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x477e)
    #2 0x7ff1623a20df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7ff162891b40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7ff16239fc3e  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x3c3e)
    #2 0x7ff1623a20df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '010-cpu-usage'… [1m[31mCrash![0m

=================================================================
==16822==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7f03a3cbcb40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7f03a37cb77e in yajl_gen_alloc (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x477e)
    #2 0x7f03a37cd0df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7f03a3cbcb40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7f03a37cac3e  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x3c3e)
    #2 0x7f03a37cd0df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '004-run-watch'… [1m[31mCrash![0m

=================================================================
==16827==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7f25bb763b40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7f25bb27277e in yajl_gen_alloc (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x477e)
    #2 0x7f25bb2740df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7f25bb763b40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7f25bb271c3e  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x3c3e)
    #2 0x7f25bb2740df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '009-disk-low-threshold-format'… [1m[31mCrash![0m

=================================================================
==16832==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7fee8ec54b40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7fee8e76377e in yajl_gen_alloc (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x477e)
    #2 0x7fee8e7650df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7fee8ec54b40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7fee8e762c3e  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x3c3e)
    #2 0x7fee8e7650df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '008-load-max-threshold-format'… [1m[31mCrash![0m
i3status: provided CPU number '1' above detected number of CPU 1

=================================================================
==16840==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7f742550bb40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7f742501a77e in yajl_gen_alloc (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x477e)
    #2 0x7f742501c0df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7f742550bb40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7f7425019c3e  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x3c3e)
    #2 0x7f742501c0df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '011-cpu-usage'… [1m[31mCrash![0m

=================================================================
==16847==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7fe2cb19eb40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7fe2cacad77e in yajl_gen_alloc (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x477e)
    #2 0x7fe2cacaf0df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7fe2cb19eb40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7fe2cacacc3e  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x3c3e)
    #2 0x7fe2cacaf0df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '021-battery-second'… [1m[31mCrash![0m

=================================================================
==16852==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7f21b1bb8b40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7f21b16c777e in yajl_gen_alloc (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x477e)
    #2 0x7f21b16c90df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7f21b1bb8b40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7f21b16c6c3e  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x3c3e)
    #2 0x7f21b16c90df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '007-cpu-usage-degraded-threshold-format'… [1m[31mCrash![0m

=================================================================
==16857==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7f8eae7cdb40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7f8eae2dc77e in yajl_gen_alloc (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x477e)
    #2 0x7f8eae2de0df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7f8eae7cdb40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7f8eae2dbc3e  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x3c3e)
    #2 0x7f8eae2de0df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '005-cpu-temp-max-threshold-format'… [1m[31mCrash![0m

=================================================================
==16862==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7fae690e7b40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7fae68bf677e in yajl_gen_alloc (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x477e)
    #2 0x7fae68bf80df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7fae690e7b40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7fae68bf5c3e  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x3c3e)
    #2 0x7fae68bf80df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '023-battery-not-charging'… [1m[31mCrash![0m

=================================================================
==16867==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7ff182d4bb40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7ff18285a77e in yajl_gen_alloc (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x477e)
    #2 0x7ff18285c0df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7ff182d4bb40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7ff182859c3e  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x3c3e)
    #2 0x7ff18285c0df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '017-wireless-down'… [1m[31mCrash![0m

=================================================================
==16872==ERROR: LeakSanitizer: detected memory leaks

Indirect leak of 576 byte(s) in 1 object(s) allocated from:
    #0 0x7f692958db40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7f692909c77e in yajl_gen_alloc (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x477e)
    #2 0x7f692909e0df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

Indirect leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7f692958db40 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb40)
    #1 0x7f692909bc3e  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x3c3e)
    #2 0x7f692909e0df  (/usr/lib/x86_64-linux-gnu/libyajl.so.2+0x60df)

SUMMARY: AddressSanitizer: 608 byte(s) leaked in 2 allocation(s).
Testing test case '002-cpu-temp'… [1m[31mCrash![0m
FAIL travis/run-tests.pl (exit status: 1)
```

<\details>

--- a/.travis.yml
+++ b/.travis.yml
@@ -35,7 +35,6 @@ addons:
     - libcap2-bin
     - pulseaudio
 script:
-  # TODO: re-enable sanitizers once issues are fixed
-  - autoreconf -fi && mkdir -p build && cd build && (../configure --disable-sanitizers || (cat config.log; false)) && make -j 8 V=1 && ldd ./i3status |& grep -q pulse || (echo "not linked against pulseaudio"; exit 1)
-  - make clean && (../configure --disable-sanitizers --disable-pulseaudio || (cat config.log; false)) && make -j 8 V=1 && ldd ./i3status |& grep -q pulse && (echo "linked against pulseaudio"; exit 1) || true
+  - autoreconf -fi && mkdir -p build && cd build && (../configure || (cat config.log; false)) && make -j 8 V=1 && ldd ./i3status |& grep -q pulse || (echo "not linked against pulseaudio"; exit 1)
+  - make clean && (../configure --enable-sanitizers --disable-pulseaudio || (cat config.log; false)) && make -j 8 V=1 && ldd ./i3status |& grep -q pulse && (echo "linked against pulseaudio"; exit 1) || true
   - make -j 8 check V=1 || (cat test-suite.log; false)

--- a/i3status.c
+++ b/i3status.c
@@ -207,6 +207,7 @@ static char *get_config_path(void) {
     config_path = resolve_tilde("~/.i3status.conf");
     if (path_exists(config_path))
         return config_path;
+    free(config_path);
     char *buf = strdup(xdg_config_dirs);
     char *tok = strtok(buf, ":");
     while (tok != NULL) {
@@ -230,7 +231,6 @@ static char *get_config_path(void) {
 
     die("Unable to find the configuration file (looked at "
         "~/.i3status.conf, $XDG_CONFIG_HOME/i3status/config, " SYSCONFDIR "/i3status.conf and $XDG_CONFIG_DIRS/i3status/config)");
-    return NULL;
 }
 
 /*
@@ -807,4 +807,6 @@ int main(int argc, char *argv[]) {
         struct timespec ts = {interval - 1 - (current_timeval.tv_sec % interval), (10e5 - current_timeval.tv_usec) * 1000};
         nanosleep(&ts, NULL);
     }
+
+    yajl_gen_free(json_gen);
 }

--- a/src/print_ddate.c
+++ b/src/print_ddate.c
@@ -173,4 +173,5 @@ void print_ddate(yajl_gen json_gen, char *buffer, const char *format, time_t t) 
     const size_t num = sizeof(placeholders) / sizeof(placeholder_t);
     buffer = format_placeholders(format, &placeholders[0], num);
     OUTPUT_FULL_TEXT(buffer);
+    free(buffer);
 }

--- a/src/print_run_watch.c
+++ b/src/print_run_watch.c
@@ -1,6 +1,7 @@
 // vim:ts=4:sw=4:expandtab
 #include <config.h>
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <yajl/yajl_gen.h>
 #include <yajl/yajl_version.h>
@@ -35,4 +36,5 @@ void print_run_watch(yajl_gen json_gen, char *buffer, const char *title, const c
 
     END_COLOR;
     OUTPUT_FULL_TEXT(buffer);
+    free(buffer);
 }

--- a/src/print_time.c
+++ b/src/print_time.c
@@ -82,4 +82,7 @@ void print_time(yajl_gen json_gen, char *buffer, const char *title, const char *
 out:
     *outwalk = '\0';
     OUTPUT_FULL_TEXT(buffer);
+    if (format_time != NULL) {
+        free(buffer);
+    }
 }

